app/models/user.rb

```diff
@@ -206,7 +207,15 @@ def self.authenticator(username = nil)
   end
 
   def self.authenticate(username, password, request = nil, options = {})
-    authenticator(username).authenticate(username, password, request, options)
+    user = authenticator(username).authenticate(username, password, request, options)
+    user.try(:link_user_to_session, request)
+
+    user
+  end
+
+  def link_user_to_session(request)
+    session_id = request.session_options[:id]
```

Comment:
Does the code need to account for `request` being `nil`?

Reply:
Yes, it should 👍